update doc auth design with simple token known issue (#883)
etcd-io/website: content/en/docs/v3.6/learning/design-auth-v3.md

There are two kinds of token types: simple and JWT. The simple token isn't designed for production use cases. Its tokens aren't cryptographically signed and servers must statefully track token-user correspondence; it is meant for development testing.  JWT tokens should be used for production deployments since it is cryptographically signed and verified. From the implementation perspective, JWT is stateless. Its token can include metadata including username and revision, so servers don't need to remember correspondence between tokens and the metadata.

> **Note :** There is a known issue [#18437](https://github.com/etcd-io/etcd/issues/18437) on simple token. Since etcdsever resolves token in the API layer and isn't guarded by linearizable check, and also simple token is stateful. So an etcd member may haven't finished processing previously authentication request before it receives next request, accordingly it will return an "invalid auth token" error back to client.

Reviewer: Slight re-wording for grammar.

```suggestion
**Note :** There is a known issue [#18437](https://github.com/etcd-io/etcd/issues/18437) with simple tokens. Within etcdsever token are resolved in the API layer, are not guarded by a linearizable check, and are also stateful. This means an etcd member may not have finished processing a previous authentication request before it receives the next request. In this situation it can return an "invalid auth token" error back to client.
```

Author: fixed grammar

> **Note :** There is a known issue [#18437](https://github.com/etcd-io/etcd/issues/18437) with simple tokens. Within etcdsever tokens are resolved in the API layer, are not guarded by a linearizable check, and are also stateful. This means an etcd member may not have finished processing a previous authentication request before it receives the next request. In this situation it can return an "invalid auth token" error back to client.

Reviewer 2: ```suggestion
**Note :** There is a known issue [#18437](https://github.com/etcd-io/etcd/issues/18437) with simple tokens. Within etcd servers, tokens are resolved at the API layer and simple tokens are stateful. The process is not protected by a linearizable check, meaning an etcd member may not have completed processing a previous authentication request before receiving the next one. In such cases, the member might return an "invalid auth token" error to the client. This issue is usually rare on a node with good network conditions but can occur if there is significant latency. As a workaround, applications can implement a retry mechanism to handle this error.

```

Reviewer 2: This comment applies to other places as well.

Author: fixed at all places
